data/shared/citizen/scripting/lua/scheduler.lua

```diff
@@ -1145,3 +1145,43 @@ end
 if not isDuplicityVersion then
 	LocalPlayer = Player(-1)
 end
+
+local requireCached = {}
+
+-- lua's require function to load script at runtime.
+-- note: for client, scripts should be added to `file` entries in the fx manifest
+-- ex script path : resource-name.dir.file
+function require(scriptPath)
+
+	if type(scriptPath) ~= 'string' then
+		error('Expected script path to be a string')
+	end
+
+	if requireCached[scriptPath] then
+		return
+	end
+
+	local dotIndex = string.find(scriptPath, '%.')
+	local resource = string.sub(scriptPath, 1, dotIndex - 1)
```

Comment:
and why are resources somehow an implied `.`? that's not even part of typical convention :/

Reply:
`.` I use dot same as `lua` require function convention.